vector/sdks/rag-chat/quickstarts/hono.mdx

```diff
@@ -0,0 +1,103 @@
+---
+title: Hono
+---
+<Card title="GitHub Repository" icon="github" href="https://github.com/upstash/rag-chat/tree/master/examples/cloudflare-workers" horizontal>
+  You can find the project source code on GitHub.
+</Card>
+This guide provides detailed, step-by-step instructions on how to use Upstash RAG Chat with Hono. You can also explore our [Cloudflare Workers - Hono example](https://github.com/upstash/rag-chat/tree/master/examples/nextjs/cloudflare-workers) for detailed, end-to-end examples and best practices.
+
+### Project Setup
+
+Create a new Hono application with `cloudflare-workers` template and install `@upstash/rag-chat` package.
+
+```shell
+npm create hono@latest my-app
+cd my-app
+npm install @upstash/rag-chat
+```
+
+### Setup Upstash Redis
+
+Create a Redis database using [Upstash Console](https://console.upstash.com) or [Upstash CLI](https://github.com/upstash/cli) and copy the `UPSTASH_REDIS_REST_URL` and `UPSTASH_REDIS_REST_TOKEN` into your `.dev.vars` file.
+
+```shell .dev.vars
+UPSTASH_REDIS_REST_URL=<YOUR_URL>
+UPSTASH_REDIS_REST_TOKEN=<YOUR_TOKEN>
+```
+
+### Setup Upstash Vector
+
+Create a Vector index using [Upstash Console](https://console.upstash.com) or [Upstash CLI](https://github.com/upstash/cli) and copy the `UPSTASH_VECTOR_REST_URL` and `UPSTASH_VECTOR_REST_TOKEN` into your `.dev.vars` file.
+
+```shell .dev.vars
+UPSTASH_VECTOR_REST_URL=<YOUR_URL>
+UPSTASH_VECTOR_REST_TOKEN=<YOUR_TOKEN>
+```
+
+### Setup QStash LLM
+
+Navigate to [QStash Console](https://console.upstash.com/qstash) and copy the `QSTASH_TOKEN` into your `.dev.vars` file.
+
+```shell .dev.vars
+QSTASH_TOKEN=<YOUR_TOKEN>
+```
+
+### Create a Hono App
+
+Update `/src/index.ts`:
+
+```typescript /src/index.ts
+import { Hono } from 'hono'
+import { RAGChat, upstash } from "@upstash/rag-chat";
+import { Index } from "@upstash/vector";
+import { Redis } from "@upstash/redis/cloudflare";
+
+const app = new Hono<{
+  Bindings: {
+    UPSTASH_REDIS_REST_URL: string;
+    UPSTASH_REDIS_REST_TOKEN: string;
+    UPSTASH_VECTOR_REST_URL: string;
+    UPSTASH_VECTOR_REST_TOKEN: string;
+    QSTASH_TOKEN: string;
+  };
+}>();
+
+app.get('/', async (c) => {
+  const ragChat = new RAGChat({
+    redis: new Redis({ token: c.env.UPSTASH_REDIS_REST_TOKEN, url: c.env.UPSTASH_REDIS_REST_URL }),
+    vector: new Index({ token: c.env.UPSTASH_VECTOR_REST_TOKEN, url: c.env.UPSTASH_VECTOR_REST_URL, cache: false }),
```

Comment:
for cache again

say that it's added because in hono example we use cloudflare, not because cache is needed in hono

also new lines for params